### PR TITLE
7-zip: 23.01, new

### PR DESCRIPTION
**Pull request by ouankou**
Topic Description
-----------------

- 7-zip: 23.01, new

Package(s) Affected
-------------------

- 7-zip: 23.01, new

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

Architectural progress for secondary ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

**Experimental Architectures**

Architectural progress for experimental ports does not impede on merging of this topic.

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: app-utils/7-zip/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,7 @@
+abinfo "Building 7-zip ..."
+cd ${SRCDIR}/CPP/7zip/Bundles/Alone2
```
Comment (MingcongBai) [inline app-utils/7-zip/autobuild/build]: Quote this path.

File: app-utils/7-zip/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=7-zip
+PKGSEC=utils
+PKGDEP="gcc-runtime glibc"
+BUILDDEP="llvm"
+PKGDES="7-Zip file archiver supporting a range of compression formats"
```
Comment (MingcongBai) [inline app-utils/7-zip/autobuild/defines]: A file archive manager with support for multiple formats

File: app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch
Comment (MingcongBai) [inline app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch:1]: Wait, why Clang?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch
Comment (ouankou) [reply app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch:1]: When using GCC, `-g` won't be effective for unknown reasons. Then, the packaging will fail later due to a lack of DEBUG build.

Using Clang can create a DEBUG build as expected. However, the `-Werror` flag triggers many compilation errors, while the same flag doesn't affect GCC. Therefore, a patch is added to remove this flag for Clang.
I've opened an issue in the upstream: https://github.com/ip7z/7zip/issues/20

Comment (ouankou) [reply app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch:1]: I just got a response from upstream. They know this issue and will fix it in the source for the next release.

File: app-utils/7-zip/autobuild/build (hunk 1 above)
Comment (ouankou) [reply app-utils/7-zip/autobuild/build]: Fixed.

File: app-utils/7-zip/autobuild/defines (hunk 2 above)
Comment (ouankou) [reply app-utils/7-zip/autobuild/defines]: Fixed.

File: app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch
Comment (MingcongBai) [reply app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch:1]: Okay, please record the above information in a `FIXME` comment.

```
FIXME: ...
...
Ref: ...
```

Comment (ouankou) [reply app-utils/7-zip/autobuild/patches/0001-fix-clang-llvm-build-errors.patch:1]: Done.

**Review by MingcongBai — APPROVED** [review]

**Comment by MingcongBai** [comment]
Great job. Just one thing left - squash all your commits (there are many "update"-type commits) and I will work on your onboarding tomorrow.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**